puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/pipelines/katello.groovy

```diff
@@ -35,10 +35,21 @@ pipeline {
 
             steps {
                 script {
-                    runCicoJobsInParallel([
-                        ['name': 'centos7 install test', 'job': "foreman-katello-${katello_version}-test"],
-                        ['name': 'centos7 upgrade test', 'job': "foreman-katello-upgrade-${katello_version}-test"]
-                    ])
+                    def pipelines = katello_server_distros.collect { os ->
+                        [
+                            name: "${os}-install",
+                            job: "foreman-katello-${katello_version}-${os}-install-test"
+                        ]
+                    }
+
+                    pipelines += katello_upgrade_server_distros.collect { os ->
+                        [
+                            name: "${os}-upgrade",
+                            job: "foreman-katello-${katello_version}-${os}-upgrade-test"
+                        ]
+                    }
```

Comment:
Can't we have a config data structure like this (syntax may not be correct):
```groovy
def pipelines = [
    'install': ['centos7'],
    'upgrade': ['centos7']
]
```

I think this makes the variables more self documenting and this pipeline code more generic. We can also introduce a helper (shared with luna) that has the signature:
```groovy
run_pipelines_in_cico(type, version, pipelines)
```

Katello would call it as `run_pipelines_in_cico('katello', katello_version, pipelines)` and luna as `run_pipelines_in_cico('luna', 'nightly', pipelines)`.

Reply:
I'll take a look at this, I don't want to merge till Monday when more devs are around in case I break things anyway.